frontend/dockerfile/dockerfile2llb/directives.go

```diff
@@ -33,12 +44,39 @@ func DetectSyntax(r io.Reader) (string, string, []parser.Range, bool) {
 }
 
 func ParseDirectives(r io.Reader) map[string]Directive {
+	bs, err := io.ReadAll(r)
```

Comment:
Random comment; do we need a limit reader for this? (Thinking of `mv latest-blockbuster.mp4 Dockerfile && docker build .`)

Reply:
I think ideally we would have the reads cached, instead of limiting how much to read here?

Idea - we could use `TeeReader` with a `bytes.Buffer` and then `io.MultiReader` to merge them together, so that how much to read is left down to the directive.

But actually, I'm not sure how much of a concern this is? The only place we call the `DetectSyntax` function from is [here](https://github.com/jedevc/buildkit/blob/directive-proposal/frontend/dockerfile/builder/build.go#L377), so it's a buffer that's "cheap" to read from (aside from the duplication cost). I suppose we could also modify the API to take raw bytes?

Reply:
I definitely haven't looked in-depth how it's used exactly 😅 - my eye fell on it, and I know we had (still have) some place in moby/moby where we just "read everything", which could be potentially dangerous if everything is read into memory (either _intentional_ (bad actor) or unintentional ("whoops! wrong file")